packages/fiori/src/SideNavigation.ts

```diff
@@ -327,24 +270,183 @@ class SideNavigation extends UI5Element {
 		return this.hasHeader && !this.collapsed;
 	}
 
-	get _itemsTree() {
-		return this.getDomRef()!.querySelector<Tree>("#ui5-sn-items-tree");
+	get hasFixedItems() {
+		return !!this.fixedItems.length;
 	}
 
-	get _fixedItemsTree() {
-		return this.getDomRef()!.querySelector<Tree>("#ui5-sn-fixed-items-tree");
+	get _rootRole() {
+		return this._inPopover ? "None" : undefined;
 	}
 
-	_walk(callback: (current: TSideNavigationItem) => void) {
-		this.items.forEach(current => {
-			callback(current);
+	get classes() {
+		return {
+			root: {
+				"ui5-sn-phone": isPhone(),
+				"ui5-sn-tablet": isTablet(),
+				"ui5-sn-combi": isCombi(),
+				"ui5-sn-collapsed": this.collapsed,
+				"ui5-sn-in-popover": this._inPopover,
+			},
+		};
+	}
 
-			current.items.forEach(currentSubitem => {
-				callback(currentSubitem);
+	getEnabledFixedItems() : Array<ITabbable> {
+		return this.getEnabledItems(this.fixedItems);
+	}
+
+	getEnabledFlexibleItems() : Array<ITabbable> {
+		return this.getEnabledItems(this.items);
+	}
+
+	getEnabledItems(items: Array<SideNavigationItem>) : Array<ITabbable> {
+		let result = new Array<ITabbable>();
+
+		items.forEach(item => {
+			if (!item.disabled) {
+				result.push(item);
+			}
+
+			if (!this.collapsed && item.expanded) {
+				result = result.concat(item.items.filter(el => !el.disabled));
+			}
+		});
+
+		return result;
+	}
+
+	focusItem(item: SideNavigationItemBase) {
+		if (item.isFixedItem) {
+			this._fixedItemNavigation.setCurrentItem(item);
+		} else {
+			this._flexibleItemNavigation.setCurrentItem(item);
+		}
+	}
+
+	onAfterRendering() {
+		const activeElement = this.shadowRoot!.activeElement;
+		const flexibleDom = this.shadowRoot!.querySelector(".ui5-sn-flexible")!;
+		if (!flexibleDom.contains(activeElement)) {
+			const selectedItem = this._findSelectedItem(this.items);
+			if (selectedItem) {
+				this._flexibleItemNavigation.setCurrentItem(selectedItem);
+			} else {
+				const focusedItem = this._findFocusedItem(this.items);
+				if (!focusedItem) {
+					this._flexibleItemNavigation.setCurrentItem(this.items[0]);
+				}
+			}
+		}
+
+		const fixedDom = this.shadowRoot!.querySelector(".ui5-sn-fixed");
+		if (!fixedDom?.contains(activeElement)) {
+			const selectedItem = this._findSelectedItem(this.fixedItems);
+			if (selectedItem) {
+				this._fixedItemNavigation.setCurrentItem(selectedItem);
+			} else {
+				const focusedItem = this._findFocusedItem(this.fixedItems);
+				if (!focusedItem) {
+					this._fixedItemNavigation.setCurrentItem(this.fixedItems[0]);
+				}
+			}
+		}
+	}
+
+	_findFocusedItem(items: Array<SideNavigationItem>) : SideNavigationItemBase | null {
+		let focusedItem = null;
+
+		if (this.collapsed) {
+			items.forEach(item => {
```

Comment:
Array.prototype.find can stop the loop once the item is found. forEach will loop through the whole length of the array.